node/src/Pos/DB/Block.hs

```diff
@@ -148,17 +150,17 @@ loadDataWhile
     -> (a -> Bool)
     -> HeaderHash
     -> m (NewestFirst [] a)
-loadDataWhile getter predicate start = NewestFirst <$> doIt start
+loadDataWhile getter predicate start = NewestFirst <$> doIt [] start
   where
-    doIt :: HeaderHash -> m [a]
-    doIt h
-        | h == genesisHash = pure []
+    doIt :: [a] -> HeaderHash -> m [a]
+    doIt !acc h
+        | h == genesisHash = pure (reverse acc)
```

Comment:
What happened here? Previously we were doing the right fold, now we're doing left fold and reverse. What for?

Reply:
It performs better.

Reply:
How? Why do you think so?

Reply:
I profiled it to verify, using it in a pure computation to get 10000 "block headers" (`Int`s). This changed dropped it from ~4ms to ~3ms. I don't know exactly why. I believe the issue was with the `fmap`s through the recursion. What we end up with is `fmap (d0 :) . fmap (d1 :) . fmap (d2 :) . ... . fmap (dn :) (pure [])`. Maybe inlining would help but it depends upon the choice of monad. And anyway, using the accumulator makes this tail recursive, from which I believe GHC can produce faster code.